Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -2977,10 +3046,85 @@ - (void)clearUnreadMessagesIndicator
     // make sure we don't show it again.
     self.hasClearedUnreadMessagesIndicator = YES;
 
-    if (self.offersAndIndicators.unreadIndicatorPosition) {
+    if (self.dynamicInteractions.unreadIndicatorPosition) {
         // If we've just cleared the "unread messages" indicator,
         // update the dynamic interactions.
-        [self ensureThreadOffersAndIndicators];
+        [self ensureDynamicInteractions];
+    }
+}
+
+- (void)viewDidLayoutSubviews
+{
+    [super viewDidLayoutSubviews];
+
+    [self updateLastVisibleTimestamp];
```

Comment:
This is to ensure that the "scroll to bottom" indicator is moved & updated appropriately when the input toolbar appears, disappears or is resized, e.g. when user input wraps to a new line and the input toolbar grows to accommodate it.